[dynamo] add meta fn for aten.kthvalue.default

ColinPeppler opened the pull request:
I saw
```
torch._dynamo.exc.Unsupported: unsupported operator: aten.kthvalue.default
```

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130562



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

ColinPeppler committed:
[dynamo] add meta fn for aten.kthvalue.default

[ghstack-poisoned]

ColinPeppler commented on torch/_meta_registrations.py:
```diff
@@ -5772,10 +5772,6 @@ def scalar_tensor(s, dtype=None, layout=None, device=None, pin_memory=None):
-        lambda: "selected index k out of range",
-    )
     sliceSize = 1 if self.dim() == 0 else self.size(dim)
     torch._check(k >= 0 and k <= sliceSize, lambda: "k not in range for dimension")
```
Found out these two `torch._checks` check the same thing. So, I removed the first one :)

ColinPeppler committed:
Update on "[dynamo] add meta fn for aten.kthvalue.default"

I saw
```
torch._dynamo.exc.Unsupported: unsupported operator: aten.kthvalue.default
```




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

jingsh reviewed (approved):
lgtm

jingsh commented:
seems some CI failed, related?

aakhundov commented:
> seems some CI failed, related?

Yeah, CI failures look legit.

ColinPeppler committed:
Update on "[dynamo] add meta fn for aten.kthvalue.default"

I saw
```
torch._dynamo.exc.Unsupported: unsupported operator: aten.kthvalue.default
```




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

ColinPeppler commented:
Also seeing this, but not sure how to proceed on this exactly. I have no issue with `topk.values`.

```
NotImplementedError: aten::kthvalue.values: attempted to run this operator with Meta tensors, but there was no fake impl or Meta kernel registered. You may have run into this message while using an operator with PT2 compilation APIs (torch.compile/torch.export); in order to use this operator with those APIs you'll need to add a fake impl. Please see the following for next steps:  https://pytorch.org/tutorials/advanced/custom_ops_landing_page.html
```

ColinPeppler commented:
Yeah, CI failures were legit but all but one were unexpected success ;)

zou3519 commented on test/test_proxy_tensor.py:
```diff
@@ -1999,7 +1999,6 @@ def f(t):
     xfail('geqrf', ''),  # aten.geqrf.default - couldn't find symbolic meta function/decomposition
     xfail('histogram', ''),  # Could not run 'aten::histogram.bin_ct' with arguments from the 'Meta' backend. This c...
     xfail('histogramdd', ''),  # aten._histogramdd_bin_edges.default - couldn't find symbolic meta function/decomposition
-    xfail('kthvalue', ''),  # aten.kthvalue.default - couldn't find symbolic meta function/decomposition
```
Add `kthvalue` to out_symbolic_tensor_failures. You added a meta for one of the two kthvalue operators but not the other one. This xfail was blocking both of those from showing up in CI

ColinPeppler committed:
Update on "[dynamo] add meta fn for aten.kthvalue.default"

I saw
```
torch._dynamo.exc.Unsupported: unsupported operator: aten.kthvalue.default
```




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

ColinPeppler commented on torch/_meta_registrations.py:
```diff
@@ -5785,6 +5781,22 @@ def topk_meta(self, k, dim=-1, largest=True, sorted=True):
 
 
+@register_meta([aten.kthvalue.default, aten.kthvalue.values])
+@out_wrapper("values", "indices")
```
Adding this helped

zou3519 replied:
nice

ColinPeppler commented:
@pytorchbot merge

ColinPeppler commented:
@pytorchmergebot merge

ColinPeppler commented:
@pytorchbot merge
